File: napakalaki(ruby)/pkg/napakalaki-0.0.1/lib/bad_consequence.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame
require_relative 'player'
require_relative 'treasure'

  
class BadConsequence
  @@MAXTREASURES = 10
  attr_reader :text, :levels
  
  def initialize(t, l)
    @text=t
    @levels=l
  end
  
  def self.getMaxTreasures
    return @@MAXTREASURES
  end
  
  def adjustToFitTreasureList(v, h)
    raise NotImplementedError.new
  end
  
  def isEmpty()
    raise NotImplementedError.new
  end
  
  def substractHiddenTreasure(t)
    raise NotImplementedError.new
  end
  
  def substractVisibleTreasure(t)
    raise NotImplementedError.new
  end

  private_class_method :new
  
  def to_s()
   " \n Texto: #{@text} \n Niveles perdidos: #{@levels} \n "
  end
  
end
end

File: napakalaki/src/Napakalaki/BadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Napakalaki;

import java.util.ArrayList;

/**
 *
 * @author javi
 */
public abstract class BadConsequence {
    static final int MAXTREASURES = 10;
    private String text;
    private int levels;
    
    public BadConsequence(String _text, int _levels){
        text=_text;
        levels=_levels;
    }
    
    public String getTtext(){
        return text;
    }
    
    public int getLevels(){
        return levels;
    }
    public abstract BadConsequence adjustToFitTreasureList(ArrayList<Treasure> v, ArrayList<Treasure> h);
    public abstract boolean isEmpty();
    public abstract void substractHiddenTreasure(Treasure t);
    public abstract void substractVisibleTreasure(Treasure t);
    
     
    
    @Override
    public String toString(){
        return "Texts = " + (text) 
                + " Levels = " + Integer.toString(levels)
                + "\n";
    }
    
}
